drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,931 @@
+}
+
+/**
+ *
+ * uart_init
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static void uart_init(uint8_t chan, uint8_t baudrate, uint8_t int_mode,
+		      uint8_t is_enable_parity, uint8_t is_event_parity,
+		      uint8_t is_onestop_bit)
+{
+	volatile struct T_ITEUART  *pComPort
+		= mComPortTb[chan].mUartBase;
+	const struct T_BaudrateStruct *pBR
+		= &mBaudrateTb[baudrate];
+	unsigned char ucTemp;
+
+	/* Enable auto gating
+	 * UART Clock Gating (UART12CG) 0: Operation,
+	 * 1: Clocks to UART1/UART2 modules are gated.
+	 */
+	CGCTRL3R &= ~(BIT(2));
+	AUTOCG   &= ~(BIT(6 - chan));
+
+	/* Enable uart module */
+	RSTDMMC  |= BIT(3 - chan);      /* Set EC side control */
+	RSTC4    = BIT(1 + chan);       /* W-One to reset controller */
+	GCR1    |= mComPortTb[chan].ucGCR1;
+	GCR6    |= mComPortTb[chan].ucGCR6;
+	GCR8    |= mComPortTb[chan].ucGCR8;
+	if (chan == t_uart_1) {
+		GCR21 &= ~(GPH1VS | GPH2VS);
+		SET_MASK(UART2_ECSPMR, BIT(1));
+#if DEF_UART2_ENABLE_CTS_RTS
+		if (mComPortTb[chan].cts) {
+			*mComPortTb[chan].cts = 0x00;
+		}
+		if (mComPortTb[chan].rts) {
+			*mComPortTb[chan].rts = 0x00;
+			SET_MASK(pComPort->UMCR, MCR_AFE);
+			ucTemp = UART2_MSR;             /* Read Clear */
+		}
+#endif  /* DEF_UART2_ENABLE_CTS_RTS */
+	} else if (chan == t_uart_0) {
+#if DEF_UART1_ENABLE_CTS_RTS
+		if (mComPortTb[chan].cts) {
+			*mComPortTb[chan].cts = 0x00;
+		}
+		if (mComPortTb[chan].rts) {
+			*mComPortTb[chan].rts = 0x00;
+			SET_MASK(pComPort->UMCR, MCR_AFE);
+			ucTemp = pComPort->UMSR;                /* Read Clear */
+		}
+#endif /* DEF_UART1_ENABLE_CTS_RTS */
+	}
+	if (mComPortTb[chan].rx) {
+		*mComPortTb[chan].rx = 0x00;
+	}
+	if (mComPortTb[chan].tx) {
+		*mComPortTb[chan].tx = 0x00;
+	}
+#if DEF_ENABLE_CTS_RTS
+	if (mComPortTb[chan].cts) {
+		*mComPortTb[chan].cts = 0x00;
+	}
+	if (mComPortTb[chan].rts) {
+		*mComPortTb[chan].rts = 0x00;
+		SET_MASK(pComPort->UMCR, MCR_AFE | MCR_RTS);
+		pComPort->URTR  = 56;
+	}
+#endif /* DEF_ENABLE_CTS_RTS */
+
+	/* set baudrate */
+#if CONFIG_ITE_UART_DMA_MODE
+	pComPort->UBRCDIR = pBR->baud_int;
+	pComPort->UBRCDDR = pBR->baud_den;
+	pComPort->UBRCDNR = pBR->baud_num;
+#endif /* CONFIG_ITE_UART_DMA_MODE */
+	SET_MASK(pComPort->ULCR, LCR_DLAB);
+	pComPort->UDLL = pBR->baud_udll;
+	pComPort->UDLM = pBR->baud_udlm;
+	pComPort->UFCR = FCR_FIFO_EN | FCR_RFRST | FCR_XFRST |
+					FCR_64B_FIFO_EN | FCR_FIFO_8;
+	pComPort->ULCR = LCR_8N1 | (is_enable_parity ? LCR_PEN : 0) |
+					(is_event_parity ? LCR_EPS : 0) |
+					(is_onestop_bit ? 0 : LCR_STB);
+	ISR4  |= BIT(6 + chan);
+	SET_MASK(pComPort->UMCR, MCR_OUT2);
+}
+
+/**
+ *
+ * get_br_idx
+ *
+ * @return
+ * STATUS_SUCCESS - input baud rate is match supported type
+ * STATUS_WARN_BAUDRATE - input baud rate isn't match, choice the ceiling
+ * @parameter
+ *
+ *
+ */
+static int get_br_idx(uint32_t baud_rate, uint32_t *baud_rate_index)
+{
+	int ret = STATUS_SUCCESS;
+
+	if (baud_rate <= 4800) {
```

Comment:
It seems to me this code would be better as a for loop and a table of baud rates